core/utils/collection.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Inspired by Laravel Collection.
+ * @link https://github.com/illuminate/collections
+ */
+namespace Elementor\Core\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Collection implements \ArrayAccess, \Countable, \IteratorAggregate {
+	/**
+	 * The items contained in the collection.
+	 *
+	 * @var array
+	 */
+	protected $items;
+
+	/**
+	 * Collection constructor.
+	 *
+	 * @param array $items
+	 */
+	public function __construct( array $items ) {
+		$this->items = $items;
+	}
+
+	/**
+	 * @param callable $callback
+	 *
+	 * @return $this
+	 */
+	public function filter( callable $callback ) {
+		if ( ! $callback ) {
```

Comment:
With the typehint it can't be null.

Reply:
Add a default null to the method. (it can be callable or null, I checked to be sure).